File: check-if-one-string-swap-can-make-strings-equal/check-if-one-string-swap-can-make-strings-equal.py
class Solution:
    def areAlmostEqual(self, s1: str, s2: str) -> bool: 
        length = len(s1) 
        count = 0

        if s1 == s2:
            return True

        if len(s1) != len(s2):
            return False

        if sorted(s1) != sorted(s2):
            return False

        for  i in range(length):
            if s1[i] != s2[i]:
                count += 1

        if count == 2:
            return True
        else: False
        

File: 1915-check-if-one-string-swap-can-make-strings-equal/check-if-one-string-swap-can-make-strings-equal.cpp
class Solution {
public:
    bool areAlmostEqual(string s1, string s2) {
        int n = s1.size();

        if (s1 == s2) {
            return true;
        }

        vector<int> diff;
        for (int i = 0; i < n; i++) {
            if (s1[i] != s2[i]) {
                diff.push_back(i);
            }
            if (diff.size() > 2) {
                return false;
            }
        }

        return diff.size() == 2 && s1[diff[0]] == s2[diff[1]] &&
               s1[diff[1]] == s2[diff[0]];
    }
};